pkg/runtime.v2/framework/plugins/torch/torch.go

```diff
@@ -43,11 +46,87 @@ func (t *Torch) Name() string {
 	return Name
 }
 
-func (t *Torch) EnforceMLPolicy(info *runtime.Info) error {
+// TODO (andreyvelich): Add support for PyTorch elastic when JobSet supports Elastic Jobs.
+func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *kubeflowv2.TrainJob) error {
 	if info == nil || info.MLPolicy == nil || info.MLPolicy.Torch == nil {
 		return nil
 	}
-	// TODO: Need to implement main logic.
+
+	// TrainJob contains the actual information for the Trainer.
+	numNodes := info.MLPolicy.NumNodes
+	if trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.NumNodes != nil {
+		numNodes = trainJob.Spec.Trainer.NumNodes
+	}
+	info.Trainer.NumNodes = numNodes
+
+	numProcPerNode := info.MLPolicy.Torch.NumProcPerNode
+	if trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.NumProcPerNode != nil {
+		numProcPerNode = trainJob.Spec.Trainer.NumProcPerNode
+	}
+
+	// Update envs for Info object.
+	// Add PyTorch distributed "PET_" values for torchrun
+	// TODO (andreyvelich): Add validation to check that TrainJob doesn't have "PET_" envs.
+	infoEnvs := []corev1.EnvVar{
+		{
+			Name:  constants.TorchEnvNumNodes,
+			Value: fmt.Sprintf("%d", *numNodes),
+		},
+		{
+			Name:  constants.TorchEnvNumProcPerNode,
+			Value: *numProcPerNode,
+		},
+		{
+			Name: constants.TorchEnvNodeRank,
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: constants.JobCompletionIndexFieldPath,
+				},
+			},
+		},
+		{
+			Name:  constants.TorchEnvMasterAddr,
+			Value: fmt.Sprintf("%v-%v-0-0.%v", trainJob.Name, constants.JobTrainerNode, trainJob.Name),
+		},
+		{
+			Name:  constants.TorchEnvMasterPort,
+			Value: fmt.Sprintf("%d", constants.ContainerTrainerPort),
+		},
+	}
+
+	// Map for all Info envs.
+	envNames := make(map[string]bool, len(infoEnvs))
+	for _, env := range infoEnvs {
+		envNames[env.Name] = true
+	}
```

Comment:
```suggestion
	envNames := sets.New[string]()
	for _, env := range infoEnvs {
        envNames.Insert(env.Name)
	}
```

Reducing maloc would be better.